.pre-commit-config.yaml

```diff
@@ -0,0 +1,40 @@
+# pre-commit 1.1.0 is required for `exclude`
+# however `minimum_pre_commit_version` itself requires 1.15.0
+minimum_pre_commit_version: "1.15.0"
+
+repos:
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
+    hooks:
+    - id: trailing-whitespace
+    - id: end-of-file-fixer
+    - id: no-commit-to-branch
+      args: [--branch, main]
```

Comment:
btw this line isn't necessary any more: "main" and "master" are both protected by default
```suggestion
```